resolwe/elastic/indices.py

```diff
@@ -406,19 +409,29 @@ def get_permissions(self, obj):
         contain list of ids of users/groups with ``view`` permission.
         """
         # TODO: Optimize this for bulk running
+        permissions_subquery = Subquery(
+            # Override the default ordering to simplify the query.
+            Permission.objects.filter(
+                content_type=ContentType.objects.get_for_model(obj),
+                codename__startswith='view',
+            ).order_by().values('pk')
+        )
+        public_user_subquery = Subquery(
+            get_user_model().objects.filter(username=ANONYMOUS_USER_NAME).values('pk')
+        )
+
         filters = {
             'object_pk': obj.id,
-            'content_type': ContentType.objects.get_for_model(obj),
-            'permission__codename__startswith': 'view',
+            'permission': permissions_subquery,
         }
         return {
             'users': list(
-                UserObjectPermission.objects.filter(**filters).distinct('user').values_list('user_id', flat=True)
```

Comment:
Don't we need `distinct` users?

Reply:
No, because we don't do any joins, so we cannot get duplicated rows.